src/types/inc/viewport.hpp

```diff
@@ -90,8 +91,8 @@ namespace Microsoft::Console::Types
             const YWalk y;
         };
 
-        bool WalkInBounds(COORD& pos, const WalkDir dir) const noexcept;
-        bool WalkInBoundsCircular(COORD& pos, const WalkDir dir) const noexcept;
+        bool WalkInBounds(COORD& pos, const WalkDir dir, bool allowBottomExclusive = false) const noexcept;
+        bool WalkInBoundsCircular(COORD& pos, const WalkDir dir, bool allowBottomExclusive = false) const noexcept;

```

Comment:
I cannot comment on the veracity of the viewport changes, so @miniksa will need to in my stead.

Reply:
OK, I read through the definitions. They look functionally correct, by my accounting.

I really do not like them. But I think I accept them for the time being.
Reasoning as:
- It sort of messes up the actual meaning of some of these functions from the external call perspective.  `InBounds` was supposed to be inside the rectangle. And now it's inside with a bit hanging off, conditionally.
- I would have greatly preferred if the alternates were either a secondary name or perhaps that the caller were responsible for checking them against `End()` if it really wanted them to be one past the last valid position that used to be `InBounds`. +/- alternate versions of `Increment()` that permit walking past the end (like an STL collection iterator end) and make it up to the caller to check where they are.
- But if that were done, then they'd probably all need helpers they implement themselves to protect safely and check both circumstances. And that sounds like a lot of busy work for the sanctity of the `Viewport` class which has honestly decayed in purity as we find more and more purposes for it.
- There's another circumstance I can think of where we probably want some region of out-of-bound areas to be valid and that's the delayed newline (which might want the entire RightExclusive() region to count as "in bounds" for walking the insertion point but not for display purposes). 
- We could theoretically encapsulate this such that you just construct a "special" version of `Viewport` by passing a regular one into a constructor with a flag that changes its entire mode. Then the `true` wouldn't have to be passed to every one of these methods by the caller who needs to work in the special domain.
- We already know that we need to probably refactor `Viewport` wholesale in the future into `til::rect` or something and de-duplicate methods that are similar, add nice `operators` and work it into a family that includes `til::size` and `til::pos`. So perhaps we just accept this as it is right now and save the "purity" of it for some future refactor date.
- It's more important, business wise, at this point to get accessibility working for us and for our partners than it is to spin extra time on solving this now. Even if it is likely making `Viewport` marginally harder to maintain going forward (versus a "more pure" solution).

So I think the action item is to capture some of this discussion/thoughts into #4015 and make it better later.